src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -171,6 +179,34 @@ public Conversion ClassifyImplicitConversionFromType(TypeSymbol source, TypeSymb
             return GetImplicitUserDefinedConversion(null, source, destination, ref useSiteInfo);
         }
 
+        /// <summary>
+        /// Helper method that calls <see cref="ClassifyImplicitConversionFromType"/> or
+        /// <see cref="HasImplicitFunctionTypeConversion"/> depending on whether the
+        /// types are <see cref="FunctionTypeSymbol"/> instances.
+        /// Used by method type inference and best common type only.
+        /// </summary>
+        public Conversion ClassifyImplicitConversionFromTypeOrImplicitFunctionTypeConversion(TypeSymbol source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var sourceFunctionType = source as FunctionTypeSymbol;
+            var destinationFunctionType = destination as FunctionTypeSymbol;
+
+            if (sourceFunctionType is null && destinationFunctionType is null)
+            {
+                return ClassifyImplicitConversionFromType(source, destination, ref useSiteInfo);
+            }
+
+            if (sourceFunctionType is { } && destinationFunctionType is { })
+            {
+                return HasImplicitFunctionTypeConversion(sourceFunctionType, destinationFunctionType, ref useSiteInfo) ?
+                    Conversion.FunctionType :
+                    Conversion.NoConversion;
+            }
+
+            Debug.Assert(false);
+            return Conversion.NoConversion;
```

Comment:
>return Conversion.NoConversion [](http://example.com/codeflow?start=12&length=30)

It is not clear why conversions to System.MulticastDelegate and the like are not checked.

Reply:
The only callers of this method are `BestTypeInferrer` and `MethodTypeInferrer` in cases where "bounds" are merged. In those cases, there are either two function types or two types that are not function types, so conversions from function type to non-function type do not need to be considered.